src/chart/drawers/chart-background.drawer.ts

```diff
@@ -9,20 +9,23 @@ import { ChartAreaTheme, FullChartConfig } from '../chart.config';
 import { getDPR } from '../utils/device/device-pixel-ratio.utils';
 import { deepEqual } from '../utils/object.utils';
 import { floor } from '../utils/math.utils';
+import { CanvasBoundsContainer } from '../canvas/canvas-bounds-container';
 
 export class BackgroundDrawer implements Drawer {
-	constructor(private canvasModel: CanvasModel, private config: FullChartConfig) {}
+	constructor(
+		private canvasModel: CanvasModel,
+		private config: FullChartConfig,
+		private canvasBoundsContainer: CanvasBoundsContainer,
+	) {}
 
 	// we need to save previous state to avoid unnecessary redraws
 	private prevState: Partial<ChartAreaTheme> = {};
-	private prevWidth = 0;
-	private prevHeight = 0;
+	private prevCanvasLocation = {};
```

Comment:
I would say that it is not clean solution.
1. The default value should be initialized to some value or undefined, in constructor for example. 
2. You compare objects by reference, which is not safe, because if some day we will decide to mutate `canvasOnPageLocation` object rather then recreate, this code will break. I think you need to compare what you need.